apps/internal/mock/mock.go

```diff
@@ -46,6 +46,13 @@ func WithCallback(callback func(*http.Request)) responseOption {
 	})
 }
 
+// WithCallback sets a callback to invoke before returning the response.
+func WithHttpStatusCode(statusCode int) responseOption {
```

Comment:
https://go.dev/wiki/CodeReviewComments#initialisms
```suggestion
func WithHTTPStatusCode(statusCode int) responseOption {
```